src/services/gas-price/gas-price.service.ts

```diff
@@ -66,22 +66,24 @@ export class GasPriceService {
   ): Promise<GasSettings> {
     let gasSettings: GasSettings = {};
 
-    const gasLimit = await contractWithSigner.estimateGas[action](
-      ...params,
-      options
-    ).catch(err => {
-      return Promise.reject(err.error);
-    });
-    gasSettings.gasLimit = this.formatGasLimit(gasLimit.toNumber());
-
-    if (this.shouldSetGasPriceSettings(options)) {
-      gasSettings = await this.setGasPriceSettings(
-        gasSettings,
-        forceLegacyTxType
+    try {
+      const gasLimit = await contractWithSigner.estimateGas[action](
+        ...params,
+        options
       );
-    }
+      gasSettings.gasLimit = this.formatGasLimit(gasLimit.toNumber());
 
-    return gasSettings;
+      if (this.shouldSetGasPriceSettings(options)) {
+        gasSettings = await this.setGasPriceSettings(
+          gasSettings,
+          forceLegacyTxType
+        );
+      }
+
+      return gasSettings;
+    } catch (err) {
+      return Promise.reject(err);
```

Comment:
Does this work exactly the same as not having a try/catch? As it's just re-throwing. I think the original implementation was trying to return err.message, but this works.

Reply:
@alter-eggo I pushed a test to showcase this scenario. You can remove the try-catch and the test will still pass. 

IMO, we should avoid this try-catch until the higher level, the one that is responsible to process the exception to translate it to a human-friendly user message or to trigger a recovery mechanism. I would also expect proper tests to cover those cases. 

For instance, in this concrete case, we have this code in a higher level: 

https://github.com/balancer-labs/frontend-v2/blob/33ff0174b9ec950ca3b525661a4902ed5006a3ba/src/services/web3/transactions/concerns/contract.concern.ts#L51

, where we are assuming that we always get a `WalletError` which is dangerous. That's outside the scope of this bugfix but I would suggest refactoring this kind of error hanndling to [something like this](https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript) (together with more new tests).

Reply:
thanks for your help here @agualis , I agree that it might be better to refactor here and created a task for this.